joomla/joomla-cms
# [4.4] Check pagination parameters from request, another try

### Summary of Changes

Existing approach still have a problem, it is always forcing the parameters defined in `$defaultUrlParams`, and make it impossible to remove. 

I suggest a different approach, take the allowed parameters from the router, and filtering them before adding. Also this now happen in the pagination constructor, which allows for developers to unset unneeded params.



### Testing Instructions

Code review. Test with your component.

Or, add following code, somewhere here https://github.com/joomla/joomla-cms/blob/67cd90ebcd3c8b12a3edd8103c0d94e3236a189c/templates/cassiopeia/index.php#L202
```php
<?php
$pg = new \Joomla\CMS\Pagination\Pagination(100, 0, 20);
$pg->setAdditionalUrlParam('tmpl', null);
echo $pg->getPaginationLinks();
?>
```
Then open any page with `?tmpl=index` in URL, and check pagination links.



### Actual result BEFORE applying this Pull Request
Somehow works.

Pagination links contain `?tmpl=index` .


### Expected result AFTER applying this Pull Request
Works but better.

Pagination links without `?tmpl=index` .


### Link to documentations
Please select:
